### PR TITLE
Add formik

### LINKED ISSUE #31: Deps: Add formik for form validation
:https://formik.org/docs/overview

### PR DESCRIPTION
**Pull request by Marwa-k**
@MuhannedNoman 
here is the register and contact us form by using formik

### CODE REVIEW

File: src/components/ContactUsForm/ContactUsForm.jsx [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+          <h2 className="text-center mt-5">GET IN TOUCH </h2>
+          <form
+            onSubmit={handleSubmit}
+            className="bg-white shadow-md rounded px-8 pt-6 pb-8 mb-4"
+          >
+            <label
+              htmlFor="email"
+              className="block text-gray-700 text-sm font-bold mb-2"
+            >
+              Email
```
Comment (MuhannedNoman) [inline src/components/ContactUsForm/ContactUsForm.jsx]: Let's use {t('Email')}

**Review by MuhannedNoman — CHANGES_REQUESTED** [review]
Let us change the text to be imported from the `i18next`
So First define the key with the same name of translation in the public folder. 

Example in en 
```
"Email": "Email"
```
In ar
```
"Email":"البريد الألكتروني"
```

Then In the contact us and registration form 

```
import { useTranslation } from 'react-i18next';


  const { t } = useTranslation();


{t('Email')}
```

**Comment by MuhannedNoman** [comment]
Also, update the snapshots they are failing due to the changes I made. 
Overall great work here :+1

**Review by NohaPepeep — APPROVED** [review]

**Review by MuhannedNoman — APPROVED** [review]
LGTM. Nice work here @Marwa-k

File: src/components/RegisterForm/RegisterForm.jsx [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+            {formik.errors.confirmPassword && (
+              <div className="text-red-500 text-sm font-bold">
+                {formik.errors.confirmPassword}
+              </div>
+            )}
+            <button
+              className="bg-red-500 mt-4 hover:bg-red-700 text-white font-bold py-2 px-4 rounded focus:outline-none focus:shadow-outline"
+              type="submit"
+            >
+              Sign UP
```
Comment (MuhannedNoman) [inline src/components/RegisterForm/RegisterForm.jsx]: This should be a t('Sign up)

File: src/components/RegisterForm/RegisterForm.jsx [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+              htmlFor="confirmPassword"
+              className="block text-gray-700 text-sm font-bold mb-2"
+            >
+              {t('confirmPassword')}
+              <input
+                className="shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline"
+                id="confirmPassword"
+                name="confirmPassword"
+                type="password"
+                placeholder="Confirm your password"
```
Comment (MuhannedNoman) [inline src/components/RegisterForm/RegisterForm.jsx]: This should be a t('')

File: src/components/RegisterForm/RegisterForm.jsx [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+              htmlFor="password"
+              className="block text-gray-700 text-sm font-bold mb-2"
+            >
+              {t('Password')}
+              <input
+                className="shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline"
+                id="password"
+                name="password"
+                type="password"
+                placeholder="Enter your password"
```
Comment (MuhannedNoman) [inline src/components/RegisterForm/RegisterForm.jsx]: Use the i18next

File: src/components/ContactUsForm/ContactUsForm.jsx [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+            {formik.errors.message && (
+              <div className="text-red-500 text-sm font-bold">
+                {formik.errors.message}
+              </div>
+            )}
+            <button
+              className="bg-red-500 mt-4 hover:bg-red-700 text-white font-bold py-2 px-4 rounded focus:outline-none focus:shadow-outline"
+              type="submit"
+            >
+              SEND
```
Comment (MuhannedNoman) [inline src/components/ContactUsForm/ContactUsForm.jsx]: Here as well

File: src/components/ContactUsForm/ContactUsForm.jsx [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+              className="block text-gray-700 text-sm font-bold mb-2"
+            >
+              {t('Message')}
+
+              <textarea
+                className="shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline"
+                id="message"
+                name="message"
+                rows="5"
+                placeholder="Enter your Message"
```
Comment (MuhannedNoman) [inline src/components/ContactUsForm/ContactUsForm.jsx]: Use i18next

File: src/components/ContactUsForm/ContactUsForm.jsx [hunk 7]
```diff
@@ -0,0 +1,74 @@
+import React from 'react';
+import { Formik } from 'formik';
+import * as Yup from 'yup';
+import { useTranslation } from 'react-i18next';
+
+export default function ContactUsForm() {
+  const { t } = useTranslation();
+  const ReviewError = Yup.object().shape({
+    email: Yup.string().email().required('Email is Required'),
```
Comment (MuhannedNoman) [inline src/components/ContactUsForm/ContactUsForm.jsx]: Here as well use i18next

**Review by MuhannedNoman — CHANGES_REQUESTED** [review]

File: src/components/RegisterForm/RegisterForm.jsx (hunk 3 above)
Comment (Marwa-k) [reply src/components/RegisterForm/RegisterForm.jsx]: @MuhannedNoman  @DhiaaAlshamy 
here I did only for placeholder

Comment (DhiaaAlshamy) [reply src/components/RegisterForm/RegisterForm.jsx]: @Marwa-k  make it for the text that appears on the screen < which is the placeholder

**Pull request merged**